src/dialects/sqlite/query-interface.js

```diff
@@ -44,11 +74,41 @@ class SQLiteQueryInterface extends QueryInterface {
     const fields = await this.describeTable(tableName, options);
     Object.assign(fields[attributeName], this.normalizeAttribute(dataTypeOrOptions));
 
-    const sql = this.queryGenerator.removeColumnQuery(tableName, fields);
-    const subQueries = sql.split(';').filter(q => q !== '');
+    const savepointName = 'change_column_savepoint';
 
-    for (const subQuery of subQueries) {
-      await this.sequelize.query(`${subQuery};`, { raw: true, ...options });
+    // Disable foreign key constraints for the next sequence of operations
+    await this.sequelize.query('PRAGMA foreign_keys = OFF;', options);
+    // Use SAVEPOINT to allow for nested transactions
+    await this.sequelize.query(`SAVEPOINT ${savepointName};`, options);
+
+    try {
+      const sql = this.queryGenerator.removeColumnQuery(tableName, fields);
```

Comment:
A lot of code here is shared with `removeColumn`, can they be deduplicated?

Reply:
I was thinking the same thing while writing the code and I had the following dilemma. 
Extracting the logic to a common function will reduce the overall code however it will increase the cognitive load due to the added abstraction. I may however rethink the way we can extract this logic to a separate function using your suggestion for the ```withForeignKeysOff``` function